source/CrystalView.mc

```diff
@@ -96,23 +87,62 @@ class CrystalView extends Ui.WatchFace {
 		mSettingsChangedSinceLastDraw = true;
 	}
 
-	function onSettingsChangedSinceLastDraw() {
+	function onSettingsChangedSinceLastDraw(dc) {
+
+		mTimezones = App.getApp().getProperty("timezones").toNumber();
+		if (mTimezones != mTimezonesOrig){
+			chooseLayout(dc);
+		} else {
+			// Recreate background buffers for each meter, in case theme colour has changed.
+			mTime.onSettingsChanged();
+			mDrawables[:LeftGoalMeter].onSettingsChanged();
+			mDrawables[:RightGoalMeter].onSettingsChanged();
+
+			mDrawables[:MoveBar].onSettingsChanged();
+			
+			if (mTimezones > 0) {
+				mDrawables[:Timezones].onSettingsChanged();
+			} 
+			// If watch does not support per-second updates, and watch is sleeping, do not show seconds immediately, as they will not 
+			// update. Instead, wait for next onExitSleep(). 
+			if (PER_SECOND_UPDATES_SUPPORTED || !mIsSleeping) { 
+				setHideSeconds(App.getApp().getProperty("HideSeconds")); 
+			}
+		}
+
+		mSettingsChangedSinceLastDraw = false;
+
+	}
+
+	function chooseLayout(dc){
 
-		// Recreate background buffers for each meter, in case theme colour has changed.
-		mDrawables[:LeftGoalMeter].onSettingsChanged();
-		mDrawables[:RightGoalMeter].onSettingsChanged();
+		if (mTimezones > 0){
+			mLayout=[];
+			mDrawables = {};
+			mTime = "";
+			setLayout(Rez.Layouts.WatchFaceTZ(dc));
+		} else {
+			mLayout=[];
+			mDrawables = {};			
+			mTime = "";
+			setLayout(Rez.Layouts.WatchFace(dc));
+		}
 
-		mDrawables[:MoveBar].onSettingsChanged();
+		mTimezonesOrig = mTimezones;
+		
+		mIconsFont = Ui.loadResource(Rez.Fonts.IconsFont);
```

Comment:
All layouts for a given watch face current use the same icon and normal fonts, so I _think_ these fonts do not need to be reloaded and re-set when the layout is changed?